src/hooks/useMediaPlayer.ts

```diff
@@ -0,0 +1,43 @@
+import { useMemo, useRef, useState } from "react";
+import { AudioManager } from "../lib/AudioManager";
+
+export type PlayState = "pause" | "play";
+
+export function useMediaPlayer() {
+  const audioManager = useMemo(() => new AudioManager(), []);
+
+  const [currentMedia, setCurrentMedia] = useState<string>("");
+  const [playState, setPlayState] = useState<PlayState>("pause");
+
+  const previousMediaLoad = useRef(Promise.resolve());
+
+  async function playMedia(fileName: string) {
+    // Wait for the previous load to finish
+    // to avoid to incur into concurrency issues
+    await previousMediaLoad.current;
```

Comment:
This solves the issue of having multiple audio loaded when doing a panic click on the song button.